### ISSUE TITLE
No module named 'torch.autograd.profiler_util' error

### ISSUE THREAD

**Issue by fosimoes** [speechbrain/speechbrain#1522]
I am getting the following error when importing speechbrain.

    import speechbrain as sb
speechbrain/__init__.py:4: in <module>
    from .core import Stage, Brain, create_experiment_directory, parse_arguments
speechbrain/core.py:35: in <module>
    from speechbrain.utils.distributed import run_on_main
speechbrain/utils/__init__.py:11: in <module>
    from . import *  # noqa
speechbrain/utils/profiling.py:11: in <module>
    from torch.autograd.profiler_util import (  # pytorch v1.10.1
E   ModuleNotFoundError: No module named 'torch.autograd.profiler_util'

The error occurs with speechbrain v0.5.12, but v0.5.11 works fine.
I tested both torch 1.10.0 and 1.11.0, with similar results.
Any idea what it might be?

**Comment by anautsch** [comment]
@fosimoes sorry to hear that - and thanks for reporting. 

We introduced profiling based on pytorch's profiler in v0.5.12 - I looked into it and tested with pytorch 1.10 and 1.11 - earlier pytorch versions didn't had `torch.autograd.profiler_util` (hence the comment); looks like they removed it afterwards again (althought a legacy pytorchh profiler library existed already with pytorch 1.11, if I remember correctly). sigh. Until now, it has caused more trouble than good, as everything changes between systems and their environment configurations (e.g., the number of nested function calls).

Wondering if it would be best to take some/all profiling features out - or - to only activate it for certain pytorch versions. Or perhaps another way...

The methods that are imported from `torch.autograd.profiler_util` are only relevant for some 'nicer' reporting. Their imports and the depending code could also be simply removed for the time being.

@mravanelli @TParcollet what are your thoughts? Would it also make sense to have a requirement check/dependency testing if functions we intend to call are still present for all sb requirements and their version ranges?

**Comment by mravanelli** [comment]
Hi Andrea,
maybe we can take the profiling test out and run them only periodically
(just like what we do with other tests like the recipe tests, etc). What do
you think?

**Comment by fosimoes** [comment]
update: 
Despite using a dockerfile based on pytorch:1.11.0, I was forcing torchaudio to v0.9.0 (due to previous conflicts), which caused pytorch to downgrade to v1.9.0.  So my tests were actually run on pytorch 1.9.0, which caused the profiler to fail.
So I changed my dockerfile to force torchaudio to v0.11:
 - pip install torchaudio==0.11.0 -f https://download.pytorch.org/whl/cu113/torch_stable.html
After that everything worked fine

**Comment by mravanelli** [comment]
I'm able to replicate the error as well when I install pytorch 1.9 (and
torchaudio 0.9). Any idea Andreas?

**Comment by anautsch** [comment]
@mravanelli testing is not the only issue here - I did too much and it became too stiff to a particular version, apparently. Sorry for that. The profiling profiles pytorch way too well -./'
No strong feelings here; your intuition some weeks back was pretty on point, ensuring the patch is pytorch future safe.

@fosimoes thank you for testing further. Despite bad news, it's valuable! I took the broken imports out of `core.utils.profiling` - and their dependencies in test_profiling - it's commented out (want to find a better solution, when it's time for that).

PR follows in a bit...
